packages/xstate-wallet/src/workflows/application.ts

```diff
@@ -144,21 +147,19 @@ const generateConfig = (
         JoinChannel: {target: 'confirmJoinChannelWorkflow'}
       }
     },
-    confirmCreateChannelWorkflow: {
-      invoke: {
-        src: 'invokeCreateChannelConfirmation',
-        onDone: {
-          target: 'createChannelInStore'
-        }
-      }
-    },
+    confirmCreateChannelWorkflow: getDataAndInvoke(
+      'getDataForCreateChannelConfirmation',
+      'invokeCreateChannelConfirmation',
+      'createChannelInStore'
+    ),
     confirmJoinChannelWorkflow: {
-      invoke: {
-        src: 'invokeCreateChannelConfirmation',
-        onDone: {
-          target: 'openChannelAndDirectFundProtocol',
-          actions: [actions.assignChannelId, actions.spawnObserver]
-        }
+      ...(getDataAndInvoke(
+        'getDataForCreateChannelConfirmation',
+        'invokeCreateChannelConfirmation'
+      ) as StateNodeConfig<WorkflowContext, {}, WorkflowEvent>),
```

Comment:
💡 We could add `WorkflowEvent` to the generic types of `getDataAndInvoke`, and avoid casting here